csharp/src/Apache.Arrow/Arrays/BooleanArray.cs

```diff
@@ -13,12 +13,125 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using Apache.Arrow.Memory;
 using Apache.Arrow.Types;
+using System;
+using System.Collections.Generic;
 
 namespace Apache.Arrow
 {
-    public class BooleanArray: PrimitiveArray<byte>
+    public class BooleanArray: Array
```

Comment:
Why is this no longer a `PrimitiveArray`?

Reply:
I get it now - because of things like this: https://issues.apache.org/jira/projects/ARROW/issues/ARROW-5035